### DISCUSSION TITLE
Boost cross-compilation

### DISCUSSION THREAD

**Discussion by Nemirtingas** [microsoft/vcpkg#20633] [Q&A] [answered]
Hi, since some new update (I don't know which one), I'm unable to cross-compile boost on Linux for Windows.
I used to modify some vcpkg's boost config files to make it cross-compile with clang but now boost explicitly calls cmd and cl which is not good at all as cl is now clang-cl and cmd is unavailable on Linux:

```
compile-c-c++ /vcpkg/buildtrees/boost-exception/x86-windows-nemirtingas-rel/boost/build/60cd9bfd1738a04fb9ae35eaaedb748c/clone_current_exception_non_intrusive.obj

    cmd.exe /S /C call "/usr/vcvarsall.bat"  ">nul" "&&"  cl "../src/clone_current_exception_non_intrusive.cpp" -Fo"/vcpkg/buildtrees/boost-exception/x86-windows-nemirtingas-rel/boost/build/60cd9bfd1738a04fb9ae35eaaedb748c/clone_current_exception_non_intrusive.obj"     -TP /EHs /GR -Xclang -fexceptions -Xclang -fcxx-exceptions -Xclang -D_CRT_SECURE_NO_WARNINGS --target=i686-windows-msvc -fms-compatibility-version=19.11 -Wno-unused-command-line-argument -imsvc /clang_windows_sdk/msvc/include -imsvc /clang_windows_sdk/winsdk/Include/10.0.18362.0/ucrt -imsvc /clang_windows_sdk/winsdk/Include/10.0.18362.0/shared -imsvc /clang_windows_sdk/winsdk/Include/10.0.18362.0/um -imsvc /clang_windows_sdk/winsdk/Include/10.0.18362.0/winrt -Xclang -ivfsoverlay -Xclang /vcpkg/buildtrees/boost-exception/x86-windows-nemirtingas-rel/winsdk_vfs_overlay.yaml /MT /O2 /Oi /Gy /DNDEBUG /Z7 /O2 /Z7 /Ob2 /W3 /MT /Ogiy /Gs /Ot /GB /Op -c -DBOOST_ALL_NO_LIB=1 -DNDEBUG "-I../include" "-I/vcpkg/installed/x86-windows-nemirtingas/include"
```

I'm trying to debug this and fix again boost build anyone got a tip in the meanwhile ?

**Comment by Nemirtingas** [comment]
It seems I have multiple issues.
1st: I need to configure boost-build correctly so it can detect my compiler/linker:
```
        compiler = [ feature.get-values <compiler> : $(options) ] ;
        compiler ?= cl ;

        linker = [ feature.get-values <linker> : $(options) ] ;
        linker ?= link ;

        resource-compiler = [ feature.get-values <resource-compiler> : $(options) ] ;
        resource-compiler ?= rc ;
```
It looks like thoses feature.get things are not filled with my compilers settings, it should be clang-cl-9, lld-link-9 and llvm-rc-9, how could I set this features ?

Also, I'd want to set rewrite-setup-scripts to off in options:  `else if <rewrite-setup-scripts>off in $(options) || [ os.name ] != NT`.


I want to change this behavior in tools/boost-build/src/tools/msvc.jam

**Comment by Nemirtingas** [comment, marked as answer]
I found the issue, vcpkg was setting up the msvc toolkit while I was using clang-cl.
Fixed it in my commit here: https://github.com/Nemirtingas/vcpkg/commit/8ef6051702454645eb671a8f62fe2c716eb10c9f

**Comment by Nemirtingas** [comment]
Does anyone use clang-cl with vcpkg and build boost ? Cause I feel like boost port is always configuring boost's msvc jam file instead of clang-win jam file ?

**Reply by Neumann-A** [reply]
I think the boost ports are currently not correctly setup for this.
